tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -83,5 +85,76 @@ private string TranslateError (string culture, string errorCode)
 			PropertyInfo propertyInfo = typeof (MSBStrings).GetProperty (errorCode);
 			return (string) propertyInfo.GetValue (null, null);
 		}
+
+		private List<string> commonIgnoreList { get; set; } = null;
+
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("pl-PL")]
+		[TestCase ("pt-BR")]
+		[TestCase ("ru-RU")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void AllErrorTranslation (string culture)
+		{
+			string errorCode = string.Empty;
+			string errorList = string.Empty;
+			bool isFirstRun = false;
+			List<string> cultureIgnoreList = null;
+			List<string> commonValidEntries = new List<string> ();
+			List<string> cultureValidEntries = new List<string> ();
+
+			string fullCommonPath = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/common-Translations.ignore";
+			string fullCulturePath = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore/{culture}-Translations.ignore";
+			string shortCommonPath = "xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/Tasktests/LocalizationIgnore/common-Translations.ignore";
+			string shortCulturePath = $"xamarin-macios/tests/msbuild/Xamarin.MacDev.Tasks.Tests/Tasktests/LocalizationIgnore/{culture}-Translations.ignore";
+			bool isCommonFileExistant = File.Exists (fullCommonPath);
+			bool isCultureFileExistant = File.Exists (fullCulturePath);
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+
+			if (commonIgnoreList == null && isCommonFileExistant){
+				commonIgnoreList = new List<string> (ReadFile (fullCommonPath));
+				isFirstRun = true;
+			}
+
+			if (isCultureFileExistant)
+				cultureIgnoreList = new List<string> (ReadFile (fullCulturePath));
+
+			foreach (var errorCodeInfo in typeof (MSBStrings).GetProperties ()) {
+				try {
+					errorCode = errorCodeInfo.Name;
+					if (errorCode == "ResourceManager" || errorCode == "Culture")
+						continue;
+					string englishError = TranslateError ("en-US", errorCode);
+					string newCultureError = TranslateError (culture, errorCode);
+
+					if (isCommonFileExistant && commonIgnoreList.Contains (errorCode)) {
```

Comment:
You don't need the isCommonFileExistant check here.
```suggestion
					if (commonIgnoreList.Contains (errorCode)) {
```